Document batch company creation endpoint (#496)
mautic/developer-documentation-new: docs/rest_api/companies.rst

.. vale off

> Create Batch Companies

Author: The batch company creation endpoint (`POST /companies/batch/new`) was added in PR #12273, which implemented duplicate detection using unique identifier fields. PR #16128 adds tests confirming the behavior documented here.

Source: https://github.com/mautic/mautic/pull/12273

> **********************

Reviewer: ```suggestion
.. vale off

Create batch Companies
**********************

.. vale on
```

.. vale on

Creates multiple Companies in a single request.

.. code-block:: php

   <?php

   $data = array(
       array(
           'companyname'    => 'Acme Corporation',
           'companyemail'   => 'info@acme.com',
           'companywebsite' => 'https://acme.com',
       ),
       array(
           'companyname'    => 'Widget Industries',
           'companyemail'   => 'info@widgetindustries.com',
           'companywebsite' => 'https://widgetindustries.com',
       ),

>    );

Reviewer 2: ```suggestion
   $data = [
       [
           'companyname'    => 'Acme Corporation',
           'companyemail'   => 'info@acme.com',
           'companywebsite' => 'https://acme.com',
       ],
       [
           'companyname'    => 'Widget Industries',
           'companyemail'   => 'info@widgetindustries.com',
           'companywebsite' => 'https://widgetindustries.com',
       ],
   ];
```

           201,
           201,
           200
       ],
       "companies": {
           "1": {
               "id": 1,
               "fields": {
                   "all": {
                       "companyname": "Acme Corporation",
                       "companyemail": "info@acme.com"
                   }
               }
           },
           "2": {
               "id": 2,
               "fields": {
                   "all": {
                       "companyname": "Widget Industries",
                       "companyemail": "info@widgetindustries.com"
                   }
               }
           },
           "3": {
               "id": 1,
               "fields": {
                   "all": {
                       "companyname": "Acme Corporation",
                       "companyemail": "info@acme.com"
                   }
               }
           }
       }
   }

Properties
----------

.. list-table::
   :widths: 25 25 50
   :header-rows: 1

   * - Name
     - Type
     - Description
   * - ``statusCodes``
     - array
     - HTTP status codes for each Company in the request: ``201`` for created, ``200`` for duplicates
   * - ``companies``
     - object

>      - A mapped collection of Companies indexed by their position in the request. Refer to :ref:`Company properties <get Company properties>` for the structure of each Company object.

Reviewer: There are a couple of things here:

- Using HTTP in uppercase is correct here. We can wrap this in vale off/on statements.
- We can make the "company" row consistent with the List Companies.

 
```suggestion
.. vale off

.. list-table::
   :widths: 25 25 50
   :header-rows: 1

   * - Name
     - Type
     - Description
   * - ``statusCodes``
     - array
     - HTTP status codes for each Company in the request: ``201`` for created, ``200`` for duplicates
   * - ``companies``
     - object
     - A mapped collection of Companies indexed by their ID
 
  For the rest of the properties, refer to :ref:`Company properties <get Company properties>`. 

.. vale on
```
